internal/demo/script.go

```diff
@@ -3,28 +3,183 @@ package demo
 import (
 	"context"
 	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
 
+	"github.com/pkg/errors"
+	"github.com/windmilleng/tilt/internal/engine"
 	"github.com/windmilleng/tilt/internal/hud"
+	"github.com/windmilleng/tilt/internal/hud/client"
 	"github.com/windmilleng/tilt/internal/k8s"
-	"github.com/windmilleng/tilt/internal/model"
+	"github.com/windmilleng/tilt/internal/logger"
+	"github.com/windmilleng/tilt/internal/output"
+	"github.com/windmilleng/tilt/internal/store"
+	"github.com/windmilleng/tilt/internal/tiltfile"
+	"golang.org/x/sync/errgroup"
+	"k8s.io/api/core/v1"
 )
 
 // Runs the demo script
 type Script struct {
 	hud   hud.HeadsUpDisplay
-	upper model.ManifestCreator
+	upper engine.Upper
+	store *store.Store
 	env   k8s.Env
+
+	readTiltfileCh chan string
+	podMonitor     *podMonitor
+}
+
+func NewScript(upper engine.Upper, hud hud.HeadsUpDisplay, env k8s.Env, st *store.Store) Script {
+	s := Script{
+		upper:          upper,
+		hud:            hud,
+		env:            env,
+		readTiltfileCh: make(chan string),
+		podMonitor:     &podMonitor{podsReadyCh: make(chan bool)},
+		store:          st,
+	}
+	st.AddSubscriber(s.podMonitor)
+	return s
+}
+
+type podMonitor struct {
+	podsReady   bool
+	podsReadyCh chan bool
+}
+
+func (m *podMonitor) OnChange(ctx context.Context, store *store.Store) {
+	state := store.RLockState()
```

Comment:
Why isn't the defer for the unlock right after this call?

Reply:
ah, i meant to move this into a helper, good catch!